src/components/views/rooms/MemberList.tsx

```diff
@@ -48,199 +30,32 @@ import { ButtonEvent } from "../elements/AccessibleButton";
 import EntityTile from "./EntityTile";
 import MemberTile from "./MemberTile";
 import BaseAvatar from "../avatars/BaseAvatar";
-import { shouldShowComponent } from "../../../customisations/helpers/UIComponents";
-import { UIComponent } from "../../../settings/UIFeature";
 import PosthogTrackers from "../../../PosthogTrackers";
-import { SDKContext } from "../../../contexts/SDKContext";
-import { canInviteTo } from "../../../utils/room/canInviteTo";
-import { inviteToRoom } from "../../../utils/room/inviteToRoom";
-import { Action } from "../../../dispatcher/actions";
-import { SpaceScopeHeader } from "./SpaceScopeHeader";
-
-const INITIAL_LOAD_NUM_MEMBERS = 30;
-const INITIAL_LOAD_NUM_INVITED = 5;
-const SHOW_MORE_INCREMENT = 100;
+import { RoomMember } from "../../../models/rooms/RoomMember";
+import { ThreePIDInvite } from "../../../models/rooms/ThreePIDInvite";
+import { useMemberListViewModel } from "../../../view-models/rooms/memberlist/MemberListViewModel";
 
 interface IProps {
     roomId: string;
-    searchQuery: string;
+    header?: ReactNode;
     onClose(): void;
-    onSearchQueryChanged: (query: string) => void;
-}
-
-interface IState {
-    loading: boolean;
-    filteredJoinedMembers: Array<RoomMember>;
-    filteredInvitedMembers: Array<RoomMember | MatrixEvent>;
-    canInvite: boolean;
-    truncateAtJoined: number;
-    truncateAtInvited: number;
+    onInviteButtonClick(roomId: string): void;
+    onThreePIDInviteClick(eventId: string): void;
```

Comment:
For callbacks specifically that are only relevant to the parent component (so if we are forwarding to the VM it will probably just be proxied to the parent) I think maybe it's open to interpretation of the MVVM implementation, I've seen done both ways. Might depend on the design of the overall navigation. I'm happy to go with whatever we align on though.